raft/server.cc

```diff
@@ -1342,6 +1377,9 @@ future<> server_impl::abort(sstring reason) {
     if (_leader_promise) {
         _leader_promise->set_exception(stopped_error(*_aborted));
     }
+    if (_tick_promise) {
+        _tick_promise->set_exception(stopped_error(*_aborted));
```

Comment:
Could this execute concurrently with `server_impl::tick`? If it could, we could potentially `set_exception` here, then `set_value` inside `tick`. (which is bad)

Reply:
Why isn't is enough to just skip entire server::tick() in case _aborted is set?

Reply:
`wait_for_next_tick()` can be called after `abort()`, there is no guarantee of the opposite

Reply:
called `check_not_aborted` from `wait_for_next_tick` instead of creating `_tick_promise` in `abort()`